File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    a1 = [1, 2, 3, 4, 5]
    a2 = [2, 1, 2, 3, 2, 4, 2, 5]
    a3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    p1, p2, p3=0,0,0
    

    for i in range(len(answers)):
        if a1[i%5] == answers[i]:
            p1+=1
        if a2[i%8] == answers[i]:
            p2+=1
        if a3[i%10] == answers[i]:
            p3+=1
            
    
    max_score = max(p1,p2,p3)
    result=[p1,p2,p3]
    for a in range(3):
        if result[a] == max_score:
            answer.append(a+1)
            
    return answer
    

File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer(3);
    vector<int> rule1={1, 2, 3, 4, 5};
    vector<int> rule2={2, 1, 2, 3, 2, 4, 2, 5};
    vector<int> rule3={3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
    for(int i=0; i<answers.size(); i++){
        if(answers[i]==rule1[i%5]) answer[0]++;
        if(answers[i]==rule2[i%8]) answer[1]++;
        if(answers[i]==rule3[i%10]) answer[2]++;
    }
    vector<int> final_answer;
    int max_v = *max_element(answer.begin(), answer.end());
    for(int i=0; i<3; i++){
        if(answer[i]==max_v){
            final_answer.push_back(i+1);
        }
    }
    return final_answer;
}